tests/date-navigator/date-navigator.spec.tsx

```diff
@@ -0,0 +1,120 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import { DateNavigator } from "../../src";
+
+describe("DateNavigator", () => {
+    const today = "2024-09-05";
+
+    beforeEach(() => {
+        jest.resetAllMocks();
+        jest.useFakeTimers().setSystemTime(new Date(today));
+    });
+
+    afterEach(() => {
+        jest.useRealTimers();
+    });
+
+    it("should render a simple date displaying the selectedDate prop without navigation arrows", () => {
+        const { rerender } = render(<DateNavigator selectedDate={today} />);
+        expect(screen.getByText("5 September 2024")).toBeVisible();
+        expect(screen.getByText("Today")).toBeVisible();
+        expect(
+            screen.queryByTestId("date-navigator-left-arrow-btn")
+        ).not.toBeInTheDocument();
+        expect(
+            screen.queryByTestId("date-navigator-right-arrow-btn")
+        ).not.toBeInTheDocument();
+
+        const tomorrow = "2024-09-06";
+        rerender(<DateNavigator selectedDate={tomorrow} />);
```

Comment:
i think we can get away with not using a `rerender` here to simplify the test implementation

these assertions can be moved to the test case below for `should render the new date if it changes on rerender` and renamed to `should render other date in full`